geom/geom/src/TGeoPatternFinder.cxx

```diff
@@ -350,7 +350,7 @@ void TGeoPatternX::UpdateMatrix(Int_t idiv, TGeoHMatrix &matrix) const
 Bool_t TGeoPatternX::IsOnBoundary(const Double_t *point) const
 {
    Double_t seg = (point[0]-fStart)/fStep;
-   Double_t diff = seg - Int_t(seg);
+   Double_t diff = seg - Long64_t(seg);
```

Comment:
probably better: Double_t diff = seg - std::floor(seg) and has to be changed in all IsOnBoundary methods, but I will do it. But the protection should not be needed, unless the point is on the moon (error condition), in which case the call to IsOnBoundary does not make sense(?)

Reply:
@pzhristov to comment on the error case. I agree `floor(seg)` is probably better.